src/yunohost/tools.py

```diff
@@ -970,6 +970,43 @@ def tools_reboot(operation_logger, force=False):
         subprocess.check_call(['systemctl', 'reboot'])
 
 
+def tools_shell(command=None):
+    """
+    Launch an (i)python shell in the YunoHost context.
+
+    This is entirely aim for development.
+    """
+
+    from yunohost.utils.ldap import _get_ldap_interface
+    ldap = _get_ldap_interface()
+
+    if command:
+        exec(command)
+        return
+
+    logger.warn("The \033[1;34mldap\033[0m interface is available in this context")
+    try:
+        from IPython import embed
+        embed()
+    except ImportError:
+        logger.warn("You don't have IPython installed, consider installing it as it is way better than the standard shell.")
+        logger.warn("Falling back on the standard shell.")
+
+        import readline  # will allow Up/Down/History in the console
+        readline  # to please pyflakes
```

Comment:
Can just use `# noqa` on the line above to avoid, I think.

Reply:
Eh it was just me moving some existing code around ... I can fix that but it's not really related to this PR, idk ...